src/games/strategy/triplea/ai/proAI/util/ProPurchaseUtils.java

```diff
@@ -124,21 +125,25 @@ public boolean canUnitsBePlaced(final List<Unit> units, final PlayerID player, f
   }
 
   public void removePurchaseOptionsByCostAndProductionAndLimits(final PlayerID player, final GameData data,
-      final List<ProPurchaseOption> purchaseOptions, final int PUsRemaining, final int remainingUnitProduction,
+      final List<ProPurchaseOption> purchaseOptions, ProResourceData resourceData, final int remainingUnitProduction,
       final List<Unit> unitsToPlace, final Map<Territory, ProPurchaseTerritory> purchaseTerritories) {
+
     for (final Iterator<ProPurchaseOption> it = purchaseOptions.iterator(); it.hasNext();) {
       final ProPurchaseOption ppo = it.next();
```

Comment:
Short variable names like "ppo" are really only appropriate for really short methods. Even then favor a variable name that clearly conveys its intent and pupose. Please write this variable name in long form to make it easier to understnad its usage later in the code.